Modernize busco tool and datamanager

bernt-matthias opened the pull request:
recently odb12 became available for prokaryotes. download would now mix now odb10 and 12. see https://gitlab.com/ezlab/busco/-/issues/784

we unfortunately have no version column .. but having it in the name and label might be OK

TODO: 

- [x] https://help.galaxyproject.org/t/forming-orthogroups-using-busco-and-access-to-busco-sequences-output/12709


FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [x] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

SaimMomin12 commented:
https://github.com/galaxyproject/tools-iuc/actions/runs/13575658591?pr=6808

SaimMomin12 commented:
Seems the mv command throws an unrecognized parameter error
ping @bernt-matthias

bernt-matthias commented:
That escaleted a bit :) 

I removed to possibility for users to download reference data on the fly. This reduces data transfer significantly (for an idea: during the few tests 720MB are downloaded) also it's better for reproducibility.

Added a data manager to manage the (dynamic) options of the tool which depend on the chosen reference data. This was hardcoded so far and likely already out of date.

bernt-matthias commented on tools/busco/busco.xml:
```diff
@@ -215,9 +246,23 @@ busco
+            </assert_stdout>
         </test>
-        <test expect_num_outputs="5">
+        <!-- <test expect_num_outputs="5">
```
```suggestion
        <test expect_num_outputs="5">
```

bernt-matthias commented on tools/busco/busco.xml:
```diff
@@ -404,7 +473,10 @@ busco
+            <assert_stdout>
+                <has_text text="BUSCO analysis done"/>
+            </assert_stdout>
+        </test> -->
```
```suggestion
        </test>
```

bernt-matthias commented on tools/busco/busco.xml:
```diff
@@ -215,9 +247,23 @@ busco
+            </assert_stdout>
         </test>
-        <test expect_num_outputs="5">
+        <!-- <test expect_num_outputs="5">
```
Mistake :)

bernt-matthias commented on tools/busco/busco.xml:
```diff
@@ -65,46 +95,37 @@ busco
+        <param type="data" name="input" format="fasta,fasta.gz" label="Sequences to analyse" help="Can be an assembled genome or transcriptome (DNA), or protein sequences from an annotated gene set."/>
+        <param name="cached_db" label="Cached database with lineage" type="select">
+            <options from_data_table="busco_database">
+                <validator message="No BUSCO database is available" type="no_options"/>
```
```suggestion
                <validator message="No BUSCO database is available - please contact your Galaxy admin." type="no_options"/>
```

bernt-matthias commented on tools/busco/busco.xml:
```diff
@@ -351,12 +412,16 @@ busco
             <conditional name="lineage">
                 <param name="lineage_mode" value="auto_detect"/>
-                <param name="auto_lineage" value="--auto-lineage"/>
+                <param name="auto_lineage" value="\-\-auto-lineage"/>
```
```suggestion
                <param name="auto_lineage" value="--auto-lineage"/>
```

SaimMomin12 commented:
Hi @bernt-matthias, is the PR good to go? Some of our project collaborators are in need of the new v12 BUSCO databases.

bernt-matthias commented:
> Hi @bernt-matthias, is the PR good to go? Some of our project collaborators are in need of the new v12 BUSCO databases.

I think so.

tbrown91 commented:
I see that now the databases have been expanded to include eukaryotes as well. Are these included in this PR? I don't know where to start looking, but can help if necessary and someone points me in the right direction

tbrown91 commented:
And for another PR, this probably also needs to be repeated for compleasm: https://github.com/galaxyproject/tools-iuc/tree/main/tools/compleasm

bernt-matthias commented:
The data manager covers a few [lineages](https://github.com/galaxyproject/tools-iuc/blob/46b9dc81f8189eada8da92204777c450145c342a/data_managers/data_manager_fetch_busco/data_manager/busco_fetcher.xml#L45) (as before) and [versions odb10 and odb12](https://github.com/galaxyproject/tools-iuc/blob/46b9dc81f8189eada8da92204777c450145c342a/data_managers/data_manager_fetch_busco/data_manager/busco_fetcher.xml#L52). 

Seems that odb12 data for eukaryotes is now released (xref: https://gitlab.com/ezlab/busco/-/issues/784#note_2366507634) https://busco-data.ezlab.org/v5/data/lineages/

bernt-matthias commented:
Since compleasm seems to use the [same data table](https://github.com/galaxyproject/tools-iuc/blob/08ce25c9b0ae236d3c0c352119de9db6b56c2925/tools/compleasm/compleasm.xml#L56C39-L56C53) as busco, the new data should become available once the datamanager ahs been run by the Galaxy admin.

bernt-matthias commented:
What certainly needs an update are the hardcoded lineages: https://github.com/galaxyproject/tools-iuc/blob/08ce25c9b0ae236d3c0c352119de9db6b56c2925/tools/compleasm/macros.xml#L12

tbrown91 commented:
> Since compleasm seems to use the [same data table](https://github.com/galaxyproject/tools-iuc/blob/08ce25c9b0ae236d3c0c352119de9db6b56c2925/tools/compleasm/compleasm.xml#L56C39-L56C53) as busco, the new data should become available once the datamanager ahs been run by the Galaxy admin.

ok, that's reassuring

tbrown91 commented:
> What certainly needs an update are the hardcoded lineages:
> 
> https://github.com/galaxyproject/tools-iuc/blob/08ce25c9b0ae236d3c0c352119de9db6b56c2925/tools/compleasm/macros.xml#L12

A fun task. The new version of compleasm at least as a `--odb` flag, so maybe there's a clever way to go about this

bernt-matthias commented:
> A fun task. The new version of compleasm at least as a `--odb` flag, so maybe there's a clever way to go about this

For this the [new `busco_database_options` data table](https://github.com/galaxyproject/tools-iuc/blob/46b9dc81f8189eada8da92204777c450145c342a/data_managers/data_manager_fetch_busco_options/tool_data_table_conf.xml.sample#L6) can be used, it stores the list of lineages that are actually available for each entry in the main `busco_database` data table. 

It can be used like this: https://github.com/galaxyproject/tools-iuc/blob/46b9dc81f8189eada8da92204777c450145c342a/tools/busco/busco.xml#L178

bernt-matthias commented:
It should also be possible to parse the odb version from the `value` column of `busco_database`

mvdbeek commented:
Attention: deployment failure!

https://github.com/galaxyproject/tools-iuc/actions/runs/14262830464

bernt-matthias commented:
Deployment should be fixed here: https://github.com/galaxyproject/tools-iuc/pull/6901
